src/webui/www/private/scripts/contextmenu.js

```diff
@@ -538,5 +541,122 @@ window.qBittorrent.ContextMenu = (function() {
         }
     });
 
+    const RssFeedContextMenu = new Class({
+        Extends: ContextMenu,
+        updateMenuItems: function() {
+            let selectedRows = window.qBittorrent.Rss.rssFeedTable.selectedRowsIds();
+            this.menu.getElement('a[href$=newSubscription]').parentNode.addClass('separator');
+            if (selectedRows.length == 0) {
+                //remove seperator on top of newSubscription entry to avoid double line line
+                this.menu.getElement('a[href$=newSubscription]').parentNode.removeClass('separator');
+                this.hideItem('update');
+                this.hideItem('markRead');
+                this.hideItem('rename');
+                this.hideItem('delete');
+                this.showItem('newSubscription');
+                this.showItem('newFolder');
+                this.showItem('updateAll');
+                this.hideItem('copyFeedURL');
+            }
+            else if (selectedRows.length == 1) {
+                if (selectedRows[0] == 0) {
+                    this.showItem('update');
+                    this.showItem('markRead');
+                    this.hideItem('rename');
+                    this.hideItem('delete');
+                    this.showItem('newSubscription');
+                    this.hideItem('newFolder');
+                    this.hideItem('updateAll');
+                    this.hideItem('copyFeedURL');
+                }
+                else if (window.qBittorrent.Rss.rssFeedTable.rows[selectedRows[0]].full_data.dataUid == "") {
+                    this.showItem('update');
+                    this.showItem('markRead');
+                    this.showItem('rename');
+                    this.showItem('delete');
+                    this.showItem('newSubscription');
+                    this.showItem('newFolder');
+                    this.hideItem('updateAll');
+                    this.hideItem('copyFeedURL');
+                }
+                else {
+                    this.showItem('update');
+                    this.showItem('markRead');
+                    this.showItem('rename');
+                    this.showItem('delete');
+                    this.showItem('newSubscription');
+                    this.hideItem('newFolder');
+                    this.hideItem('updateAll');
+                    this.showItem('copyFeedURL');
+                }
+            }
+            else {
+                this.showItem('update');
+                this.showItem('markRead');
+                this.hideItem('rename');
+                this.showItem('delete');
+                this.hideItem('newSubscription');
+                this.hideItem('newFolder');
+                this.hideItem('updateAll');
+                this.showItem('copyFeedURL');
+            }
+        }
```

Comment:
This is really hard to understand. Can you document what each of these conditionals means?